packages/core/src/index.mts

```diff
@@ -36,7 +35,7 @@ type HookStore = {
 
 const hookStorage = new AsyncLocalStorage<HookStore>();
 
-const context = {
+export const context = {
```

Comment:
I specifically do not want to expose the hooks internals outside the core package. I think you're exporting it to reuse in the new utils files you created; but the side effect is exposing it through the npm package too.

If we need to do this in this PR, let's do a refactor only that'll extract the hooks code to a separate file first.

Reply:
Oops, I didn't expect to get this part merged in - this branch is more WIP. I'm aware of the side effect.

That's an excellent suggestion; I'd like to open that PR separately right away. It would be nice to rebase with main once that's merged in.

Reply:
I've opened a separate PR for this in #1292 - I will rebase my changes once that gets merged in.